pkg/sidecar/sidecar.go

```diff
@@ -10,25 +10,40 @@ import (
 	"net"
 	"os"
 	"os/exec"
-	"os/signal"
 	"path"
 	"strings"
 	"sync/atomic"
-	"syscall"
 	"time"
 
 	"github.com/andres-erbsen/clock"
 	proto "github.com/spiffe/go-spiffe/proto/spiffe/workload"
 	"github.com/spiffe/spire/api/workload"
+	"golang.org/x/sys/unix"
 )
 
-// sidecar is the component that consumes the Workload API and renews certs
+// Config contains config variables when creating a SPIFFE Sidecar.
+type Config struct {
+	AgentAddress       string `hcl:"agentAddress"`
+	Cmd                string `hcl:"cmd"`
+	CmdArgs            string `hcl:"cmdArgs"`
+	CertDir            string `hcl:"certDir"`
+	SvidFileName       string `hcl:"svidFileName"`
+	SvidKeyFileName    string `hcl:"svidKeyFileName"`
+	SvidBundleFileName string `hcl:"svidBundleFileName"`
+	RenewSignal        string `hcl:"renewSignal"`
+	Timeout            string `hcl:"timeout"`
+	ExternalProcess    bool
```

Comment:
Perhaps since ExternalProcess can't be set without ProcessReloadFunc (or the library will panic) and ProcessReloadFunc is meaningless without External process being set, maybe we should just get rid of the ExternalProcess bool? and maybe rename ProcessReloadFunc to `ReloadExternalProcess`?

Reply:
(where we determine whether or not to manage the external process by whether or not `ReloadExternalProcess` is set)

Reply:
done